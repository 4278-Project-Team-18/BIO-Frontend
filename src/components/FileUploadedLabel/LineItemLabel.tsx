import styles from "./LineItemLabel.module.css";
import {
  LineItemLabelVariant,
  type LineItemLabelProps,
} from "./LineItemLabel.definitions";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";

const LineItemLabel = ({ label, variant, icon }: LineItemLabelProps) => (
  <div
    className={
      variant === LineItemLabelVariant.GREEN
        ? styles["label-green"]
        : variant === LineItemLabelVariant.RED
        ? styles["label-red"]
        : variant === LineItemLabelVariant.YELLOW
        ? styles["label-yellow"]
        : variant === LineItemLabelVariant.BLUE
        ? styles["label-blue"]
        : variant === LineItemLabelVariant.PURPLE
        ? styles["label-purple"]
        : variant === LineItemLabelVariant.ORANGE
        ? styles["label-orange"]
        : styles["label-blue"]
    }
  >
    {icon && <FontAwesomeIcon icon={icon} />}
    <span className={styles["label"]}>{` ${label}`}</span>
  </div>
);

export default LineItemLabel;
