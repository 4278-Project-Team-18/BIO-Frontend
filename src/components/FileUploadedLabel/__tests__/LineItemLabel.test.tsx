import LineItemLabel from "../LineItemLabel";
import { LineItemLabelVariant } from "../LineItemLabel.definitions";
import {
  faCheckCircle,
  faXmarkCircle,
} from "@fortawesome/free-solid-svg-icons";
import { faker } from "@faker-js/faker";
import { render } from "@testing-library/react";

describe("LineItemLabel Tests", () => {
  test("LineItemLabel should render correctly", () => {
    faker.seed(2);

    // render sucess variant
    const successComponent = render(
      <LineItemLabel
        label={"Success Label"}
        variant={LineItemLabelVariant.GREEN}
        icon={faCheckCircle}
      />,
    );
    expect(successComponent).toBeTruthy();
    expect(successComponent.getByText("Success Label")).toBeTruthy();

    // render error variant
    const errorComponent = render(
      <LineItemLabel
        label={"Error Label"}
        variant={LineItemLabelVariant.RED}
        icon={faXmarkCircle}
      />,
    );
    expect(errorComponent).toBeTruthy();
    expect(errorComponent.getByText("Error Label")).toBeTruthy();
  });

  test("LineItemLabel snapshot should render correctly", () => {
    faker.seed(2);

    // render sucess variant
    const successComponent = render(
      <LineItemLabel
        label={"Success Label"}
        variant={LineItemLabelVariant.GREEN}
        icon={faCheckCircle}
      />,
    );
    expect(successComponent).toMatchSnapshot();

    // render error variant
    const errorComponent = render(
      <LineItemLabel
        label={"Error Label"}
        variant={LineItemLabelVariant.RED}
        icon={faXmarkCircle}
      />,
    );
    expect(errorComponent).toMatchSnapshot();
  });
});
