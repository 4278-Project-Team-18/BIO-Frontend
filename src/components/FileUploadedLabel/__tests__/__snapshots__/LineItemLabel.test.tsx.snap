// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LineItemLabel Tests LineItemLabel snapshot should render correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="label-green"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-circle-check "
          data-icon="circle-check"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M256 512A256 256 0 1 0 256 0a256 256 0 1 0 0 512zM369 209L241 337c-9.4 9.4-24.6 9.4-33.9 0l-64-64c-9.4-9.4-9.4-24.6 0-33.9s24.6-9.4 33.9 0l47 47L335 175c9.4-9.4 24.6-9.4 33.9 0s9.4 24.6 0 33.9z"
            fill="currentColor"
          />
        </svg>
        <span
          class="label"
        >
           Success Label
        </span>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="label-green"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-circle-check "
        data-icon="circle-check"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M256 512A256 256 0 1 0 256 0a256 256 0 1 0 0 512zM369 209L241 337c-9.4 9.4-24.6 9.4-33.9 0l-64-64c-9.4-9.4-9.4-24.6 0-33.9s24.6-9.4 33.9 0l47 47L335 175c9.4-9.4 24.6-9.4 33.9 0s9.4 24.6 0 33.9z"
          fill="currentColor"
        />
      </svg>
      <span
        class="label"
      >
         Success Label
      </span>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`LineItemLabel Tests LineItemLabel snapshot should render correctly 2`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="label-green"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-circle-check "
          data-icon="circle-check"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M256 512A256 256 0 1 0 256 0a256 256 0 1 0 0 512zM369 209L241 337c-9.4 9.4-24.6 9.4-33.9 0l-64-64c-9.4-9.4-9.4-24.6 0-33.9s24.6-9.4 33.9 0l47 47L335 175c9.4-9.4 24.6-9.4 33.9 0s9.4 24.6 0 33.9z"
            fill="currentColor"
          />
        </svg>
        <span
          class="label"
        >
           Success Label
        </span>
      </div>
    </div>
    <div>
      <div
        class="label-red"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-circle-xmark "
          data-icon="circle-xmark"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M256 512A256 256 0 1 0 256 0a256 256 0 1 0 0 512zM175 175c9.4-9.4 24.6-9.4 33.9 0l47 47 47-47c9.4-9.4 24.6-9.4 33.9 0s9.4 24.6 0 33.9l-47 47 47 47c9.4 9.4 9.4 24.6 0 33.9s-24.6 9.4-33.9 0l-47-47-47 47c-9.4 9.4-24.6 9.4-33.9 0s-9.4-24.6 0-33.9l47-47-47-47c-9.4-9.4-9.4-24.6 0-33.9z"
            fill="currentColor"
          />
        </svg>
        <span
          class="label"
        >
           Error Label
        </span>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="label-red"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-circle-xmark "
        data-icon="circle-xmark"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M256 512A256 256 0 1 0 256 0a256 256 0 1 0 0 512zM175 175c9.4-9.4 24.6-9.4 33.9 0l47 47 47-47c9.4-9.4 24.6-9.4 33.9 0s9.4 24.6 0 33.9l-47 47 47 47c9.4 9.4 9.4 24.6 0 33.9s-24.6 9.4-33.9 0l-47-47-47 47c-9.4 9.4-24.6 9.4-33.9 0s-9.4-24.6 0-33.9l47-47-47-47c-9.4-9.4-9.4-24.6 0-33.9z"
          fill="currentColor"
        />
      </svg>
      <span
        class="label"
      >
         Error Label
      </span>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
